acceptancetests/jujupy/client.py

```diff
@@ -583,6 +584,12 @@ def get_version_string_parts(version_string):
             return version_parts
 
 
+def strip_build_number(version_str):
+    # strip build number of the version string.
+    # for example, 2.9.27.888 -> 2.9.27
+    return '.'.join(version_str.split('.')[:3])
```

Comment:
does this work for MAJ.MIN only? like `2.9` or `2.9-beta1`

Reply:
works for everything

Reply:
actually, strip_build_number should probably got before the split('-')

Reply:
right, we have to cover the case of edge build with the OfficialBuild number like 2.9-beta1.888-ubuntu-amd64